index.d.ts

```diff
@@ -0,0 +1,490 @@
+interface AwaitingTwo<A, B, R> {
+  (a: A, b: B): R
+  (a: A): (b: B) => R
+}
+
+interface AwaitingThree<A, B, C, R> {
+  (a: A, b: B, c: C): R
+  (a: A, b: B): (c: C) => R
+  (a: A): AwaitingTwo<B, C, R>
+}
```

Comment:
...and they may be problematic: https://github.com/sanctuary-js/sanctuary/pull/431#discussion_r130345580 :(